internal/handlers/pg/pgdb/collections.go

```diff
@@ -174,16 +175,13 @@ func DropCollection(ctx context.Context, tx pgx.Tx, db, collection string) error
 		return ErrSchemaNotExist
 	}
 
-	table := formatCollectionName(collection)
-	tables, err := tables(ctx, tx, db)
+	metadata := newMetadata(tx, db, collection)
+	table, err := metadata.getTableName(ctx)
 	if err != nil {
 		return lazyerrors.Error(err)
 	}
-	if !slices.Contains(tables, table) {
-		return ErrTableNotExist
-	}
 
-	err = newMetadata(tx, db, collection).remove(ctx)
+	err = metadata.remove(ctx)
```

Comment:
I suggest removing metadata after we drop table, like in this https://github.com/FerretDB/FerretDB/pull/2224/files PR. It's not super critical, as we are still in the same transaction, but I think this approach is a bit more accurate.

Reply:
Updated it